src/reporter/tests/test_notify.py

```diff
@@ -313,3 +313,50 @@ def test_no_value_in_notification(notification):
     url = '{}'.format(notify_url)
     r = requests.post(url, data=json.dumps(notification), headers=HEADERS_PUT)
     assert r.status_code == 200
+
+    
+def test_no_value_for_attributes(notification):
+    # with empty value
+    notification['data'][0] = {
+        'id': '299531',
+        'type': 'AirQualityObserved',
+        'p': {'type': 'string', 'value': '', 'metadata': {}},
+        'pm10': {'type': 'string', 'value': '', 'metadata': {}},
+        'pm25': {'type': 'string', 'value': '', 'metadata': {}},
+    }
+    url = '{}'.format(notify_url)
+    get_url = "{}/entities/299531".format(QL_URL)
+    url_new = '{}'.format(get_url)
+    r = requests.post(url, data=json.dumps(notification), headers=HEADERS_PUT)
+    assert r.status_code == 200
+    res_get = requests.get(url_new, headers=HEADERS_PUT)
+    assert res_get.status_code == 404
+    # entity with missing value string
+    notification['data'][0] = {
+        'id': '299531',
+        'type': 'AirQualityObserved',
+        'p': {'type': 'string'},
+        'pm10': {'type': 'string', 'value': '', 'metadata': {}},
+        'pm25': {'type': 'string', 'value': '', 'metadata': {}},
+    }
+    url = '{}'.format(notify_url)
+    get_url = "{}/entities/299531/attrs/p/value".format(QL_URL)
+    url_new = '{}'.format(get_url)
+    r = requests.post(url, data=json.dumps(notification), headers=HEADERS_PUT)
+    assert r.status_code == 200
+    res_get = requests.get(url_new, headers=HEADERS_PUT)
+    assert res_get.status_code == 404
+    # entity has both valid and empty attributes
+    notification['data'][0] = {
+        'id': '299531',
+        'type': 'AirQualityObserved',
+        'p': {'type': 'string'},
+        'pm10': {'type': 'string', 'value': '10', 'metadata': {}},
+    }
+    url = '{}'.format(notify_url)
+    get_url = "{}/entities/299531/attrs/p/value".format(QL_URL)
+    url_new = '{}'.format(get_url)
+    r = requests.post(url, data=json.dumps(notification), headers=HEADERS_PUT)
+    assert r.status_code == 200
+    res_get = requests.get(url_new, headers=HEADERS_PUT)
+    assert res_get.status_code == 404
```

Comment:
Do we get a `404` here cos the attribute isn't there or the whole entity is **not** in the DB? Off the top of my head I can't tell what QL does if you query an attribute of an entity that's **not** in the DB. So could it be we get a `404` because entity `299531` wasn't inserted? Can you please check that?

What we want to check with this test case is that the entity gets inserted with a `pm10` attribute value of `10`.  So I'd just check that too, e.g.

    pm10_url = "{}/entities/299531/attrs/pm10/value".format(QL_URL)
    res_get = requests.get(pm10_url, headers=HEADERS_PUT)
    assert res_get.status_code == 200
    assert get_value(res_get) == '10'

`get_value` is a made up method, can't remember how to extract the returned value from the response :-0

Reply:
Hie @c0c0n3 , I have updated TestCase for Both valid entity and empty value string.
https://github.com/smartsdk/ngsi-timeseries-api/blob/6ff27a7aae6b50cc423bda4acc5723e5c9782639/src/reporter/tests/test_notify.py#L363
https://github.com/smartsdk/ngsi-timeseries-api/blob/6ff27a7aae6b50cc423bda4acc5723e5c9782639/src/reporter/tests/test_notify.py#L365

In this it will check response to validate that the entity gets inserted with a pm10 attribute value of 10 or not using method res_get.json().

Reply:
Good stuff! Thanks a stack!